src/diffusers/models/attention.py

```diff
@@ -202,15 +231,28 @@ def __init__(
                 if self.use_ada_layer_norm
                 else nn.LayerNorm(dim, elementwise_affine=norm_elementwise_affine, eps=norm_eps)
             )
-            self.attn2 = Attention(
-                query_dim=dim,
-                cross_attention_dim=cross_attention_dim if not double_self_attention else None,
-                heads=num_attention_heads,
-                dim_head=attention_head_dim,
-                dropout=dropout,
-                bias=attention_bias,
-                upcast_attention=upcast_attention,
-            )  # is self-attn if encoder_hidden_states is none
+            if attention_type == "temporal":
```

Comment:
Just use `Attention` here.

Reply:
As per the state_dict keys of the motion sdxl model, it looks like every attention layer has its own pos_embed layer instead of it being a part of the transformer block. Making it `Attention` causes mismatched and unexpected key errors.